onnxruntime/contrib_ops/cpu/cdist.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "assert.h"
+namespace onnxruntime {
+namespace contrib {
+
+// Computes the squared Euclidean distance between the vectors.
+template <typename T>
+class Sqeuclidean {
+ public:
+  T operator()(const T* a1, const T* b1, size_t n) const {
+    T sum = 0;
+    for (size_t k = 0; k != n; ++k) {
```

Comment:
I don't know if the compiler can optimize everything but it would be faster with: for(;n>0;--n, ++a1, ++a2) { t = *a2 - *a1; sum += t*t;}